April Finance Update

efecarranza opened the pull request:
# Changelog

Note: Includes BGDs reimbursements.

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:
```diff
@@ -0,0 +1,89 @@
+
+- Implementation: [AaveV2Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol), [AaveV2Polygon](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol), [AaveV3Gnosis](https://github.com/bgd-labs/aave-proposals-v3/bl ...
+- Tests: [AaveV2Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.t.sol), [AaveV2Polygon](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.t.sol), [AaveV3Gnosis](https://github.com/bgd-labs/aave-proposals-v3/blob/ma ...
+- [Snapshot](TODO)
```
```suggestion
```

ianflexa commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.t.sol:
```diff
@@ -0,0 +1,294 @@
+    vm.createSelectFork(vm.rpcUrl('mainnet'), 19706815);
+    proposal = new AaveV2Ethereum_AprilFinanceUpdate_20240421();
+  }
+
```
```suggestion

function test_defaultProposalExecution() public {
    defaultTest(
      'AaveV2Ethereum_AprilFinanceUpdate_20240421',
      AaveV2Ethereum.POOL,
      address(proposal)
    );
  }
  
```

ianflexa commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.t.sol:
```diff
@@ -0,0 +1,152 @@
+    vm.createSelectFork(vm.rpcUrl('polygon'), 56325958);
+    proposal = new AaveV2Polygon_AprilFinanceUpdate_20240421();
+  }
+
```
```suggestion

function test_defaultProposalExecution() public {
    defaultTest(
      'AaveV2Polygon_AprilFinanceUpdate_20240421',
      AaveV2Polygon.POOL,
      address(proposal)
    );
  }

```

ianflexa commented on src/20240421_Multi_AprilFinanceUpdate/AaveV3Gnosis_AprilFinanceUpdate_20240421.t.sol:
```diff
@@ -0,0 +1,39 @@
+    vm.createSelectFork(vm.rpcUrl('gnosis'), 33562797);
+    proposal = new AaveV3Gnosis_AprilFinanceUpdate_20240421();
+  }
+
```
```suggestion

function test_defaultProposalExecution() public {
    defaultTest(
      'AaveV3Gnosis_AprilFinanceUpdate_20240421',
      AaveV3Gnosis.POOL,
      address(proposal)
    );
  }
  
```

ianflexa reviewed (changes requested):
missing e2e tests

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol:
```diff
@@ -0,0 +1,227 @@
+
+  function _agdAllowance() internal {
+    AaveV3Ethereum.COLLECTOR.approve(AaveV2EthereumAssets.USDT_A_TOKEN, AGD_MULTISIG, 0);
+    AaveV3Ethereum.COLLECTOR.approve(
+      AaveV3EthereumAssets.GHO_UNDERLYING,
+      AGD_MULTISIG,
+      AGD_GHO_ALLOWANCE
+    );
```
@efecarranza I would add an if condition here, to approve the GHO amount only when the allowance of aUSDT to AGD is 612.9k as if claimed mistakenly before the payload execution, the transfer could be done twice.

efecarranza replied:
@brotherlymite sorry say that again, i don't think i understood.

brotherlymite replied:
```
if (USDT_A_TOKEN.allowance(COLLECTOR, AGD_MULTISIG) >= ~612.9e6) {
  AaveV3Ethereum.COLLECTOR.approve(
      AaveV3EthereumAssets.GHO_UNDERLYING,
      AGD_MULTISIG,
      AGD_GHO_ALLOWANCE
    );
}
```
@efecarranza if the AGD wallet claims aUSDT before proposal execution, we still approve the GHO token. So there is a double approval. So just an extra check.

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:
```diff
@@ -0,0 +1,89 @@
+
+This proposal cancels the existing USDT Allowance to the Aave Grants DAO and replaces it with a GHO Allowance of equal size, 612.9k. This represents a continuation of transistioning to using GHO as the preferred means of covering DAO expenses.
+
+Also within this proposal, there is the corective action of returning [3504 RealT WXDAI](https://gnosisscan.io/tx/0xf70a507847da2ad4acbecc35cd84c2f5d2489b0b0eb0e16af49c9262ca95707e) sent by RealT to the Aave Gnosis Chain collector by accident. Details on the agreement with realT can be found [here](https://snapshot.org/#/aave.eth/proposal/0xff69be7580614ebc1a455591c1bd651d8f0af12070d277d7d8846beb ...
```
```suggestion
Also within this proposal, there is the corrective action of returning [3504 RealT WXDAI](https://gnosisscan.io/tx/0xf70a507847da2ad4acbecc35cd84c2f5d2489b0b0eb0e16af49c9262ca95707e) sent by RealT to the Aave Gnosis Chain collector by accident. Details on the agreement with realT can be found [here](https://snapshot.org/#/aave.eth/proposal/0xff69be7580614ebc1a455591c1bd651d8f0af12070d277d7d8846beb3c7c964b).
```

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:
```diff
@@ -0,0 +1,89 @@
+
+With the upcoming extension of [Merit](https://governance.aave.com/t/arfc-merit-is-forever-reward-system-program-extension/17336) and the funding request from [Chaos Labs](https://governance.aave.com/t/arfc-chaos-labs-engagement-amendment/17324), this proposal will swap sufficient assets to GHO to ensure enough funds are available as required.
+
+This proposal cancels the existing USDT Allowance to the Aave Grants DAO and replaces it with a GHO Allowance of equal size, 612.9k. This represents a continuation of transistioning to using GHO as the preferred means of covering DAO expenses.
```
```suggestion
This proposal cancels the existing USDT Allowance to the Aave Grants DAO and replaces it with a GHO Allowance of equal size, 612.9k. This represents a continuation of transitioning to using GHO as the preferred means of covering DAO expenses.
```

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:
```diff
@@ -0,0 +1,89 @@
+| Withdraw & Swap to GHO | Migrate Eth v2 to v3 |    Pol to Eth     | Migrate Pol v2 to v3 |
+| :--------------------: | :------------------: | :---------------: | :------------------: |
+|    aEthUSDC (1.4M)     |    awBTC (All-1)     |  aPolDAI (All-1)  |    amUSDT (All-1)    |
+|      aDAI (All-1)      |    awETH (All-1)     |  amUSDC (All-1)   |                      |
```
I would put an approximate amount instead of (All - 1)

efecarranza replied:
we've been using this terminology in all these proposals, if it's okay i'd rather keep this way.

brotherlymite replied:
okay

efecarranza commented:
@brotherlymite @ianflexa 

ian, i added the tests but when they yield empty files we don't include them usually.
harsh, updated but have one question.

thanks

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:
```diff
@@ -0,0 +1,184 @@
+      IERC20(AaveV3PolygonAssets.WMATIC_UNDERLYING).balanceOf(address(this))
+    );
+
+    (bool ok, ) = address(plasmaBridge).call{value: address(this).balance}('');
+
+    if (!ok) revert FailedToSendMATIC();
+  }
```
```suggestion
```
@efecarranza I would not send funds to the plasma bridge, as we will do the MATIC bridging in second part.

Luigy-Lemon replied:
Just one remark, Polygon plasma bridge has a log cap protection of 10 logs in a bridge transaction request.  Transferring funds to the bridge contract in Part A makes sense to me to avoid this issue during Part B.  

Currently not sure if it will be a problem or not since this is the first time we are doing it in production.

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:
```diff
@@ -0,0 +1,184 @@
+  IAavePolEthERC20Bridge public constant bridge =
+    IAavePolEthERC20Bridge(MiscPolygon.AAVE_POL_ETH_BRIDGE);
+
+  IAavePolEthPlasmaBridge public constant plasmaBridge =
+    IAavePolEthPlasmaBridge(0xc980508cC8866f726040Da1C0C61f682e74aBc39);
```
```suggestion
```

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:
```diff
@@ -0,0 +1,184 @@
+  function withdraw(uint256 amount) external;
+}
+
+interface IAavePolEthPlasmaBridge {
+  function bridge(uint256 amount) external;
+}
```
```suggestion
```

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:
```diff
@@ -0,0 +1,184 @@
+    );
+  }
+
+  function _withdrawMatic() internal {
```
not needed now, as the we would do the matic bridging in the second part

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:
```diff
@@ -0,0 +1,181 @@
+    }
+  }
+
+  receive() external payable {}
```
```suggestion
```
@efecarranza This is not needed as `address(this)` is the executor in this case which will receive the MATIC token.

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate_20240421.s.sol:
```diff
@@ -0,0 +1,132 @@
+
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actionsTwo = new IPayloadsControllerCore.ExecutionAction[](1);
+    actions[0] = GovV3Helpers.buildAction(payload1);
```
```suggestion
    actionsTwo[0] = GovV3Helpers.buildAction(payload1);
```

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate_20240421.s.sol:
```diff
@@ -0,0 +1,132 @@
+
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actionsPolygonTwo = new IPayloadsControllerCore.ExecutionAction[](1);
+    actionsPolygon[0] = GovV3Helpers.buildAction(
```
```suggestion
    actionsPolygonTwo[0] = GovV3Helpers.buildAction(
```

ianflexa commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:
```diff
@@ -0,0 +1,92 @@
+|      aDAI (All-1)      |    awETH (All-1)     |  amUSDC (All-1)   |                      |
+|     aUSDC (All-1)      |    aLINK (All-1)     | aPolMATIC (All-1) |                      |
+|   aUSDT (All-0.65M)    |     aSNX (All-1)     |   amDAI (All-1)   |                      |
+|                        |                      |  aPolUSDC(All-1)  |                      |
```
```suggestion
|   aEthDAI (All-1)      |                      |  aPolUSDC(All-1)  |                      |
```

brotherlymite commented on src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:
```diff
@@ -0,0 +1,92 @@
+
+Deposit wETH and wBTC in the Treasury into Aave v3.
+
+Assets migrated from Polygon to Ethereum are to be deposited into Aave v3 where practical, with the exception of DAI and USDC which will be swapped to GHO during Part B. MATIC bridging will also happen on part B.
```
```suggestion
Assets migrated from Polygon to Ethereum are to be deposited into Aave v3 where practical, with the exception of DAI and USDC which will be swapped to GHO during Part B.
```

ianflexa reviewed (commented):
missing aEthDAI
